lab-ronoh/lib/task-crud.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+// const debug = require('debug')('ronito: task-crud');
+const mongoose = require('mongoose');
+mongoose.Promise= require('bluebird');
+const Task = require('../model/task');
+const noteCrud = require('./note-crud');
+const AppError = require('./app-error');
+
+exports.createTask = function(reqBody){
+  return new Promise((resolve, reject)=> {
+    if(!reqBody.noteId)
+      return reject(AppError.error400('task requires noteId'));
+    if(!reqBody.description)
+      return reject(AppError.error400('task requires a description'));
+
+    const task = new Task(reqBody);
+    task.save()
+  .then(task => resolve(task))
```

Comment:
you can just do this like 
```javascript
task.save()
  .then(resolve)
  .catch(reject)
```
Or even 
```javascript
task.create(reqBody)
  .then(resolve)
  .catch(reject)
```
Since resolve and reject are functions, and since you're not doing anything to change the arguments from what .save() passes back, you can do it this way. You could even just get rid of the whole 'return new promise' thing and do something like 

```javascript
return task.save()
```
which would do exactly the same thing.